api/tools/build.gradle

```diff
@@ -50,49 +59,26 @@ buildscript {    // Configuration for building
   }
 }
 
+configurations {
+  compile.extendsFrom project(':').configurations.compile
+  runtime.extendsFrom project(':').configurations.runtime
+}
 
 sourceSets {
   main {
     java {
-       srcDirs = ['src/main/java', '../src/main/java', '../src/generated/java']
```

Comment:
**A** change here is needed, not necessarily this one though. My new command line uses the generated swagger notebooks client, which brings in a bunch of transitive deps which weren't handled already. I spent an hour or so trying to add these new deps here piecemeal, but ultimately couldn't satisfy the compiler. What's currently here is basically me giving up and importing everything from the main project.

I'll learn more about Gradle, but in order to improve this I'd first need to understand conceptually why we think there is a clean dependency graph of tools -> (lib) <- server, what goes in "lib", and whether what remains in "server" would be heavy enough to justify the added complexity of two build paths. My first guess would be that controllers/interceptors/server-specific stuff and their deps would go into "server", which doesn't seem like much.

Reply:
OK - I've reverted to the whitelist-based approach we had previous. Found the right include/exclude combination to get this compiling, and added okhttp.

Learned a good deal more about Gradle, but I think improving the current setup is going to require larger changes. A project structure which could eventually be good here is:

api/
- swagger
- tools

Where tools and api can cleanly depend on the "swagger" subproject. All swagger-related deps could then be defined there. Then there's the issue of whitelisting subdirs and excluding files to avoid transitive deps. This could perhaps be formalized by another subproject which somehow defines the source splits into two targets, or else we could refactor some of the logic/classes which aren't suitable for library use out of those subdirs.